www/js/theme_editor.js

```diff
@@ -98,18 +154,25 @@ const imageInput = () =>
           }
 
           try {
-            const files = await imagemin([evt.target.files[0].path], userBackgroundsPath);
-            if (files) {
-              store.setUserPref('app.themebg', {
-                type: 'custom',
-                url: `${files[0].path}`.replace(/(\s)/g, '\\ '),
-              });
-              analytics.trackEvent('theme', 'custom');
-
-              global.core.platformMethod('updateSettings');
+            const filePath = evt.target.files[0].path;
+            if (imageCheck(filePath)) {
+              const files = await imagemin([filePath], userBackgroundsPath);
+              if (files) {
+                store.setUserPref('app.themebg', {
+                  type: 'custom',
+                  url: `${files[0].path}`.replace(/(\s)/g, '\\ '),
+                });
+                recentBgsContainer.appendChild(recentSwatchFactory(files[0].path));
```

Comment:
seems like I can create a new swatch even though i've uploaded the image before. luckily it doesn't duplicate it in the directory. 
![image](https://user-images.githubusercontent.com/2982784/51789477-06f63b00-213e-11e9-8944-a8fa2a411f25.png)